homeassistant/components/overkiz/water_heater_entities/hitachi_dhw.py

```diff
@@ -63,7 +64,10 @@ def target_temperature(self) -> float | None:
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
```

Comment:
Are we sure this case can happen?

Reply:
Yes, that can happen; another valid service call would be with an `temp_high` and `temp_low` parameter, in that case, `temperature` will be missing.

Reply:
Looks like we never implemented handling high and low target temps in the water heater integration, only for climate.

https://github.com/home-assistant/core/blob/c2d372dd13bd0892f6fd3f4386efac6f7b5ce418/homeassistant/components/water_heater/__init__.py#L94-L102

It looks like it's required with that schema.

Reply:
Oh 🤦 Sorry, water heater that is my oversight, I looked at climate. That indeed requires it.

Reply:
I'm not sure to follow here, What should I do here?

Reply:
Ok, I can do it.

but in this case, Should I also do it in:
- [Demo](https://github.com/home-assistant/core/blob/dev/homeassistant/components/demo/water_heater.py#L85)
- [Econet](https://github.com/home-assistant/core/blob/dev/homeassistant/components/econet/water_heater.py#L124) (which does a check)
- [an other ovekiz entity](https://github.com/home-assistant/core/blob/dev/homeassistant/components/overkiz/water_heater_entities/hitachi_dhw.py) (an other one exists but its already correct)
- [Tado](https://github.com/home-assistant/core/blob/dev/homeassistant/components/tado/water_heater.py#L238) (which does a check)
- [Vicare](https://github.com/home-assistant/core/blob/dev/homeassistant/components/vicare/water_heater.py#L183) (which does a check)

(the list should be complete, I've searched for `kwargs.get(ATTR_TEMPERATURE)` in non climate entities)

Reply:
And if I understood correctly what was said about climate, the temperature should be checked in async_set_temperature. 
I found some places where it's not.  
Juste one exemple on [modbus](https://github.com/home-assistant/core/blob/dev/homeassistant/components/modbus/climate.py#L173)  
The result of this code could be, well, I don't know, a negative value?
Would you like me to fill a bug specically for that?

Reply:
I did it for the 2 overkiz water heater entities

Reply:
PRs for other integrations are always welcome but they need to be tested before merging.